pom.client.xml

```diff
@@ -94,7 +94,7 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <packageOutputDirectory>${project.build.directory}</packageOutputDirectory>
+    <packageOutputDirectory>${artifactStagingDirectory}/${project.artifactId}</packageOutputDirectory>
```

Comment:
Don't we need to default "artifactStagingDirectory"? If we don't then I expect local builds might fail.

Reply:
Yeah good idea. At this point I'm just experimenting. I'm not certain this is the best approach. It might be better, and more generic to have a reusable PowerShell fragment that groups by regex match and then reuse it across all languages. Thoughts?

The other idea I am toying with is having a seperate artifact per file group.

Reply:
I'd like to eliminate the regex matching if we can it is just a source of errors. It would certainly better to do have that only in this one place as opposed to the way we have it now but I think we can do better. I believe overriding the output path from the individual package build steps for the given language is better. That way the package output from the real build is the one ensuring we get the correct set of files into the correct output directory instead of a magic regex.

Reply:
That's fair, in that case this is probably the right approach then. So I'll add a default property value here that we can then override to repoint things, and people can of course still override the packageOuputDirectory value if they want to squash them together.

Obviously this is just the first in a bunch of changes that would need to be made to have this work across the ES.

Reply:
I've adjusted my approach a bit. In my latest commit I've added a profile to ```pom.client.xml``` to enable partitioning of the artifacts. My intention here is that we'll only do this as part of the Azure Pipelines builds.

I will probably need to do the same thing to pom.data.xml as well. After my POM restructuring changes I can probably just move it into pom.xml and it'll cover both scenarios.

Reply:
Ok if you do it with the profile activation then you should revert back to the old packageOutputDirectory property value by default then correct?

Reply:
Reverted